lib/net.c

```diff
@@ -515,22 +524,76 @@ lua_init_iface(struct gatekeeper_if *iface, const char *iface_name,
 		strcpy(iface->pci_addrs[i], pci_addrs[i]);
 	}
 
-	for (i = 0; i < num_ip_addrs; i++) {
-		int gk_type = get_ip_type(ip_addrs[i]);
+	for (i = 0; i < num_ip_cidrs; i++) {
+		/* Need to make copy to tokenize. */
+		char ip_cidr_copy[strlen(ip_cidrs[i]) + 1];
+		char *ip_addr;
+
+		char *prefix_len_str;
+		char *end;
+		long prefix_len;
+		int gk_type;
+
+		strncpy(ip_cidr_copy, ip_cidrs[i], strlen(ip_cidrs[i]) + 1);
+
+		ip_addr = strtok(ip_cidr_copy, "/");
+		if (ip_addr == NULL)
+			goto name;
+
+		gk_type = get_ip_type(ip_addr);
 		if (gk_type == AF_INET &&
-				inet_pton(AF_INET, ip_addrs[i],
+				inet_pton(AF_INET, ip_addr,
 				&iface->ip4_addr) == 1) {
 			iface->configured_proto |= GK_CONFIGURED_IPV4;
-			continue;
 		}
 		else if (gk_type == AF_INET6 &&
-				inet_pton(AF_INET6, ip_addrs[i],
+				inet_pton(AF_INET6, ip_addr,
 				&iface->ip6_addr) == 1) {
 			iface->configured_proto |= GK_CONFIGURED_IPV6;
-			continue;
 		}
 		else
 			goto name;
+
+		prefix_len_str = strtok(NULL, "\0");
+		if (prefix_len_str == NULL)
+			goto name;
+
+		prefix_len = strtol(prefix_len_str, &end, 10);
+		if (prefix_len_str == end || !*prefix_len_str || *end) {
+			RTE_LOG(ERR, GATEKEEPER,
+				"net: prefix length \"%s\" is not a number\n",
+				prefix_len_str);
+			goto name;
+		}
+		if ((prefix_len == LONG_MAX || prefix_len == LONG_MIN) &&
+				errno == ERANGE) {
+			RTE_LOG(ERR, GATEKEEPER,
+				"net: prefix length \"%s\" caused underflow or overflow\n",
+				prefix_len_str);
+			goto name;
+		}
+		if (prefix_len < 0 || prefix_len > max_prefix_len(gk_type)) {
+			RTE_LOG(ERR, GATEKEEPER,
+				"net: prefix length \"%s\" is out of range\n",
+				prefix_len_str);
+			goto name;
+		}
+
+		if (gk_type == AF_INET) {
+			iface->ip4_mask.s_addr =
+				rte_cpu_to_be_32(~0 << (32 - prefix_len));
+		} else if (gk_type == AF_INET6) {
+			uint64_t *paddr = (uint64_t *)iface->ip6_mask.s6_addr;
+			if (prefix_len <= 64) {
+				paddr[0] = rte_cpu_to_be_64(
+					~0 << (64 - prefix_len));
```

Comment:
Unless you properly identify otherwise through a cast or use of qualifiers, the `0` is an `int`, so whenever `(64 - prefix_len) >= 32` (i.e. `prefix_len <= 32`) this expression won't work. Moreover, once you make the `0` an `unint64_t)`, the code will still break when `prefix_len == 0`. To understand all that is going on, see my previous comment.